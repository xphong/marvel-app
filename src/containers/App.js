import React, { Component, PropTypes } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Spinner from '../components/Spinner';
import * as CountriesActions from '../actions/CountriesActions';
import Countries from '../components/Countries';

export class App extends Component {
  componentDidMount() {
    this.props.actions.fetchCountries('https://restcountries.eu/rest/v1/all');
  }

  render() {
    const { countries, actions } = this.props;
    
    return (
      <div className="mainContainer container">
        <h1>React Redux Starter</h1>
        {countries.isLoading ? <Spinner /> : <Countries countries={countries} actions={actions} />}
      </div>
    );
  }
}

App.propTypes = {
  countries: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    countries: state.countries
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(CountriesActions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
