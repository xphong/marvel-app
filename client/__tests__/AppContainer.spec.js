import React from 'react';
import ReactDOM from 'react-dom';

import App from '../containers/App';

describe('App Container', () => {
  it('should render App', () => {
    const wrapper = shallow(
      <App />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
