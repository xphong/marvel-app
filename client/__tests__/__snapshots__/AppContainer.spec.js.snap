// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container should render App 1`] = `
<div>
  <Nav />
  <div
    className="ui main container"
  />
</div>
`;
